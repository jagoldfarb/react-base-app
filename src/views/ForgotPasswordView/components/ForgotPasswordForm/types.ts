import { IForgotPasswordFormData } from 'views/ForgotPasswordView/types';

export interface IForgotPasswordFormProps {
  onSubmit: (formData: IForgotPasswordFormData) => void;
}
