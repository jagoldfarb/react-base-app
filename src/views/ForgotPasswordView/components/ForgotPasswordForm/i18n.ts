import i18next from 'i18next';
import { locales } from 'configs/i18n';

i18next.addResources(locales.EN, 'FORGOT_PASSWORD_FORM', {
  EMAIL: 'Email',
  SUBMIT_BUTTON: 'Reset my password',
});
