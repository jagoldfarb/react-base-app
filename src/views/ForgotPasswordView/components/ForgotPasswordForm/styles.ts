import styled from 'styled-components';

export const Form = styled.form`
  width: 100%;
`;

export const ButtonContainer = styled.div`
  margin-bottom: 30px;
  display: flex;
  justify-content: center;
`;
