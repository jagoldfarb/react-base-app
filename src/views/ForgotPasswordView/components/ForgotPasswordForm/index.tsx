import React, { FunctionComponent } from 'react';
import { Field, InjectedFormProps, reduxForm } from 'redux-form';
import i18next from 'i18next';

import { IForgotPasswordFormData } from 'views/ForgotPasswordView/types';
import { emailValidation, requiredValidation } from 'helpers/validations';

import Input from 'components/inputs/Input';
import Button from 'components/commons/Button';

import { IForgotPasswordFormProps } from './types';
import { Form, ButtonContainer } from './styles';
import './i18n';

const ForgotPasswordForm: FunctionComponent<
  IForgotPasswordFormProps & InjectedFormProps<IForgotPasswordFormData, IForgotPasswordFormProps>
> = (props) => {
  const { handleSubmit, onSubmit, submitting } = props;

  return (
    <Form onSubmit={handleSubmit(onSubmit)} data-testid="forgot-password-form">
      <Field
        component={Input}
        label={i18next.t('FORGOT_PASSWORD_FORM:EMAIL')}
        name="email"
        validate={[requiredValidation, emailValidation]}
        type="email"
      />
      <ButtonContainer>
        <Button disabled={submitting} type="submit">
          {i18next.t('FORGOT_PASSWORD_FORM:SUBMIT_BUTTON')}
        </Button>
      </ButtonContainer>
    </Form>
  );
};

export default reduxForm<IForgotPasswordFormData, IForgotPasswordFormProps>({
  form: 'forgotPassword',
})(ForgotPasswordForm);
