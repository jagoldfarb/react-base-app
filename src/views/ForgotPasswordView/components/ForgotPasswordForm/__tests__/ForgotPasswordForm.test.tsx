import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { reduxForm } from 'redux-form';

import store from 'redux/store';

import ForgotPasswordForm from '..';

const defaultProps = {
  onSubmit: jest.fn(),
};

const Form = reduxForm({
  form: 'forgotPassword',
})((props: any) => <ForgotPasswordForm {...props} />);

describe('ForgotPassword Form', () => {
  it('Should render the forgot password form empty', () => {
    const { asFragment, getByTestId } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Form {...defaultProps} />
        </MemoryRouter>
      </Provider>,
    );
    expect(getByTestId('forgot-password-form')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render the ForgotPassword form filled and press submit', () => {
    const props = {
      ...defaultProps,
      initialValues: {
        email: 'johndoe@gmail.com',
      },
    };
    const { asFragment, getByTestId } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Form {...props} />
        </MemoryRouter>
      </Provider>,
    );
    expect(getByTestId('forgot-password-form')).toBeDefined();
    const button = getByTestId('button');
    expect(button).toBeDefined();
    fireEvent.click(button);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render the ForgotPassword form filled with wrong fields', () => {
    const props = {
      ...defaultProps,
      initialValues: {
        email: 'johndoegmail.com',
      },
    };
    const { asFragment, getByTestId } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Form {...props} />
        </MemoryRouter>
      </Provider>,
    );
    expect(getByTestId('forgot-password-form')).toBeDefined();
    const button = getByTestId('button');
    expect(button).toBeDefined();
    fireEvent.click(button);
    expect(asFragment()).toMatchSnapshot();
  });
});
