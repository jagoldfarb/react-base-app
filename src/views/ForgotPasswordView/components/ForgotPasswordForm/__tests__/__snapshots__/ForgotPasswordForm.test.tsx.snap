// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword Form Should render the ForgotPassword form filled and press submit 1`] = `
<DocumentFragment>
  <form
    class="sc-eCssSg jvctCF"
    data-testid="forgot-password-form"
  >
    <div
      class="sc-bdfBwQ cgTScR"
    >
      <div
        class="MuiFormControl-root MuiTextField-root sc-gsTCUz jKtOdX"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink="true"
        >
          Email
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="input-Email"
            maxlength="255"
            name="email"
            rows="3"
            type="email"
            value="johndoe@gmail.com"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
            >
              <span>
                Email
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="sc-jSgupP evxpKV"
    >
      <button
        class="sc-hKgILt bytOKm"
        data-testid="button"
        type="submit"
      >
        Reset my password
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`ForgotPassword Form Should render the ForgotPassword form filled with wrong fields 1`] = `
<DocumentFragment>
  <form
    class="sc-eCssSg jvctCF"
    data-testid="forgot-password-form"
  >
    <div
      class="sc-bdfBwQ cgTScR"
    >
      <div
        class="MuiFormControl-root MuiTextField-root sc-gsTCUz jKtOdX"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-error Mui-error MuiFormLabel-filled"
          data-shrink="true"
        >
          Email
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"
        >
          <input
            aria-invalid="true"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="input-Email"
            maxlength="255"
            name="email"
            rows="3"
            type="email"
            value="johndoegmail.com"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-11 PrivateNotchedOutline-legendNotched-12"
            >
              <span>
                Email
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="sc-dlfnbm bOmIgN"
        data-testid="error-block"
      >
        Invalid email address
      </div>
    </div>
    <div
      class="sc-jSgupP evxpKV"
    >
      <button
        class="sc-hKgILt bytOKm"
        data-testid="button"
        type="submit"
      >
        Reset my password
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`ForgotPassword Form Should render the forgot password form empty 1`] = `
<DocumentFragment>
  <form
    class="sc-eCssSg jvctCF"
    data-testid="forgot-password-form"
  >
    <div
      class="sc-bdfBwQ cgTScR"
    >
      <div
        class="MuiFormControl-root MuiTextField-root sc-gsTCUz jKtOdX"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Email
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="input-Email"
            maxlength="255"
            name="email"
            rows="3"
            type="email"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                Email
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="sc-jSgupP evxpKV"
    >
      <button
        class="sc-hKgILt bytOKm"
        data-testid="button"
        type="submit"
      >
        Reset my password
      </button>
    </div>
  </form>
</DocumentFragment>
`;
